mne/realtime/tests/test_fieldtrip_client.py

```diff
@@ -22,33 +24,95 @@
 warnings.simplefilter('always')  # enable b/c these tests throw warnings
 
 
-def _run_buffer(kill_signal, neuromag2ft_fname):
+def _run_buffer(kill_signal):
+    neuromag2ft_fname = op.realpath(op.join(os.environ['NEUROMAG2FT_ROOT'],
+                                    'neuromag2ft'))
     # Works with neuromag2ft-3.0.2
     cmd = (neuromag2ft_fname, '--file', raw_fname, '--speed', '4.0')
 
+    # sleep to make sure everything is setup before playback starts
     process = subprocess.Popen(cmd, stdout=subprocess.PIPE,
                                stderr=subprocess.PIPE)
     # Let measurement continue for the entire duration
-    kill_signal.get(timeout=10.0)
+    kill_signal.get(timeout=40.0)
     process.terminate()
     with warnings.catch_warnings(record=True):  # still running
         process.stderr.close()
         process.stdout.close()
         del process
 
 
+def _start_buffer_thread():
+    signal_queue = queue.Queue()
+    thread = threading.Thread(target=_run_buffer, args=(signal_queue,))
+    thread.daemon = True
+    thread.start()
+    return signal_queue
+
+
+@pytest.mark.slowtest
+@requires_neuromag2ft
+def test_fieldtrip_rtepochs():
+    raw_tmax = 20
+    raw = mne.io.read_raw_fif(raw_fname, preload=True)
+    raw.crop(tmin=0, tmax=raw_tmax)
+    events_offline = mne.find_events(raw, stim_channel='STI 014')
+    event_id = list(np.unique(events_offline[:, 2]))
+    tmin, tmax = -0.2, 0.5
+    epochs_offline = mne.Epochs(raw, events_offline, event_id=event_id,
+                                tmin=tmin, tmax=tmax)
+    epochs_offline.drop_bad()
+    isi_max = (np.max(np.diff(epochs_offline.events[:, 0])) /
+               float(raw.info['sfreq'])) + 7.0
+
+    kill_signal = _start_buffer_thread()
+
+    try:
```

Comment:
why is this necessary?

Reply:
to clean up the ``neuromag2ft`` process in ``finally`` (instead of relying on the timeout)